src/resolve/resolved-link.c

```diff
@@ -674,14 +674,35 @@ int link_update(Link *l) {
         return 0;
 }
 
+static bool link_has_link_local_dns(Link *l, int family) {
+
+        /* Check if the link has a link-local dns server for the specified family */
+
+        LIST_FOREACH(servers, s, l->dns_servers)
```

Comment:
Hmm, bit late, but how about the global dns servers with interface specifier?

Reply:
yeah, i'd ignore global dns servers with scope. probably ignore it for good, but definitely for now.